geode-gfsh/src/main/java/org/apache/geode/management/internal/cli/commands/StatusRedundancyCommand.java

```diff
@@ -0,0 +1,121 @@
+import org.apache.geode.management.internal.operation.RebalanceOperationPerformer;
+import org.apache.geode.management.internal.security.ResourceOperation;
+import org.apache.geode.security.ResourcePermission;
+
+public class StatusRedundancyCommand extends GfshCommand {
+  static final String COMMAND_NAME = "status redundancy";
+  private static final String COMMAND_HELP =
+      "Report the redundancy status for partitioned regions in connected members. The default is to report status for all regions.";
+  static final String INCLUDE_REGION = "include-region";
+  private static final String INCLUDE_REGION_HELP =
+      "Partitioned regions to be included when reporting redundancy status. Includes take precedence over excludes.";
+  static final String EXCLUDE_REGION = "exclude-region";
+  private static final String EXCLUDE_REGION_HELP =
+      "Partitioned regions to be excluded when reporting redundancy status.";
+
+  @CliCommand(value = COMMAND_NAME, help = COMMAND_HELP)
+  @CliMetaData(relatedTopic = {CliStrings.TOPIC_GEODE_DATA, CliStrings.TOPIC_GEODE_REGION})
+  @ResourceOperation(resource = ResourcePermission.Resource.DATA,
+      operation = ResourcePermission.Operation.READ)
+  public ResultModel execute(
+      @CliOption(key = INCLUDE_REGION, help = INCLUDE_REGION_HELP) String[] includeRegions,
+      @CliOption(key = EXCLUDE_REGION, help = EXCLUDE_REGION_HELP) String[] excludeRegions) {
+    RedundancyCommandUtils utils = getUtils();
+
+    List<RebalanceOperationPerformer.MemberPRInfo> membersForEachRegion = new ArrayList<>();
+    List<String> includedRegionsWithNoMembers = new ArrayList<>();
+
+    utils.populateLists(membersForEachRegion, includedRegionsWithNoMembers, includeRegions,
+        excludeRegions, (InternalCache) getCache());
+
+    for (RebalanceOperationPerformer.MemberPRInfo prInfo : membersForEachRegion) {
+      // Filter out any members using older versions of Geode
+      List<DistributedMember> viableMembers =
+          utils.filterViableMembersForVersion(prInfo, REDUNDANCY_COMMAND_ADDED_VERSION);
+
+      if (viableMembers.size() == 0) {
+        // If no viable members were found, return with error status
+        return utils.getNoViableMembersResult(REDUNDANCY_COMMAND_ADDED_VERSION, prInfo.region);
+      } else {
+        // Update the MemberPRInfo with the viable members
+        prInfo.dsMemberList = viableMembers;
+      }
+    }
+
+    List<CliFunctionResult> functionResults =
+        executeFunctionOnMembers(includeRegions, excludeRegions, membersForEachRegion);
+
+    return utils.buildResultModelFromFunctionResults(functionResults, includedRegionsWithNoMembers,
+        true);
+  }
+
+  List<CliFunctionResult> executeFunctionOnMembers(String[] includeRegions, String[] excludeRegions,
+      List<RebalanceOperationPerformer.MemberPRInfo> membersForEachRegion) {
+
+    List<CliFunctionResult> functionResults = new ArrayList<>();
+    boolean shouldNotReassignPrimaries = false;
```

Comment:
Unneeded variable.

Reply:
Agreed. I've removed the redundant variables and introduced a comment to clarify the meaning of each of the function arguments instead.